rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,270 @@
+                  // "autoprefixer" plugin improves browser compatibility by generating vendor prefixes.
+                  // https://www.npmjs.com/package/postcss-loader
+                  loader: require.resolve('postcss-loader'),
+                  options: {
+                    postcssOptions: {
+                      plugins: [
+                        // https://www.npmjs.com/package/autoprefixer
+                        autoprefixer
+                      ]
+                    },
+
+                    sourceMap: !production
+                  }
+                },
+                {
+                  // Compiles SASS syntax into CSS
+                  // https://www.npmjs.com/package/sass-loader
+                  loader: require.resolve('sass-loader'),
+
+                  options: {
+                    implementation: sass,
+                    sassOptions: {
+                      includePaths: [path.resolve(__dirname, 'node_modules')]
+                    },
+
+                    sourceMap: !production
+                  }
+                }
+              ]
+            }
+          ]
+        },
+
+        {
+          test: /\.(jpeg|jpg|png|gif|svg|ico)$/,
+          // Allows import/require() to be used with an asset file. The file will be copied to the output folder,
+          // and the import statement will return its URL.
+          // https://www.npmjs.com/package/file-loader
+          loader: 'file-loader'
+        }
+      ]
+    },
+    devServer: {
+      host: 'localhost',
+      port: 8080
+    },
+    // See here for documentation: https://webpack.js.org/configuration/devtool
+    devtool: production ? undefined : 'eval-source-map',
+    optimization: {
+      minimize: !!production,
+      nodeEnv: production ? 'production' : 'development',
+      splitChunks: {
+        chunks: 'all'
+      },
+      minimizer: [
+        new CssMinimizerPlugin({
+          minimizerOptions: {
+            preset: ['default', { discardComments: { removeAll: !!production } }]
+          }
+        }),
+        // This magic constant indicates the minimizer defaults
+        // https://webpack.js.org/configuration/optimization/#optimizationminimizer
+        '...'
+      ]
+    },
+    plugins: [
+      // See here for documentation: https://webpack.js.org/plugins/mini-css-extract-plugin/
+      new MiniCssExtractPlugin({
+        filename: '[name].[contenthash].css'
+      }),
+
+      // See here for documentation: https://webpack.js.org/plugins/define-plugin/
+      new DefinePlugin({
+        DEBUG: !production
+      }),
+
+      process.env.REPORT ? new BundleAnalyzerPlugin({ analyzerMode: 'static' }) : undefined
+    ]
+  };
+
+  let result = mergeWithCustomize({
+    // Remove duplicate of HtmlWebpackPlugin
+    // this allows projects to override the default rig behavior if applicable (ex. when targeting older browsers)
+    customizeArray: unique(
+      'plugins',
+      ['HtmlWebpackPlugin'],
```

Comment:
(This is slightly awkward -- is there a better way?)

Reply:
I'm starting to think we should invent a more rigorous inheritance mechanism for **webpack.config.js** than these copy+paste `createWebpackConfig()` functions. When we tackle that, we can also figure out a better way to handle these overrides.